pkg/engine/plugins.go

```diff
@@ -233,38 +234,49 @@ func computeDefaultProviderPlugins(languagePlugins, allPlugins pluginSet) map[to
 			continue
 		}
 
-		if seenVersion, has := defaultProviderVersions[tokens.Package(p.Name)]; has {
-			if seenVersion == nil {
+		if seenPlugin, has := defaultProviderPlugins[tokens.Package(p.Name)]; has {
+			if seenPlugin.Version == nil {
 				logging.V(preparePluginLog).Infof(
 					"computeDefaultProviderPlugins(): plugin %s selected for package %s (override, previous was nil)",
 					p, p.Name)
-				defaultProviderVersions[tokens.Package(p.Name)] = p.Version
+				defaultProviderPlugins[tokens.Package(p.Name)] = p
 				continue
 			}
 
 			contract.Assertf(p.Version != nil, "p.Version should not be nil if sorting is correct!")
-			if p.Version != nil && p.Version.GT(*seenVersion) {
-				logging.V(preparePluginLog).Infof(
-					"computeDefaultProviderPlugins(): plugin %s selected for package %s (override, newer than previous %s)",
-					p, p.Name, seenVersion)
-				defaultProviderVersions[tokens.Package(p.Name)] = p.Version
-				continue
+			contract.Assertf(p.Version.GTE(*seenPlugin.Version),
+				"Should not have seen an older plugin if sorting is correct!\n  %s-%s %s\n  %s-%s - %s",
+				p.Name, p.Version.String(), p.Path,
+				seenPlugin.Name, seenPlugin.Version.String(), seenPlugin.Path)
+
+			if p.Version.EQ(*seenPlugin.Version) {
+				return nil, fmt.Errorf("two plugins had the same name and version.\n  %s-%s %s\n  %s-%s - %s",
+					p.Name, p.Version.String(), p.Path,
+					seenPlugin.Name, seenPlugin.Version.String(), seenPlugin.Path)
 			}
```

Comment:
the block above is new.  technically when the assert was firing before, it might have been because one plugin was less than *or* equal to another plugin.  I want to at least check if maybe it's the 'eq' case that is causing us to fire.  I sitll don't see how that could happen (as prior code shoudl have made it we dont' ever double count eq plugins.  But this will at least provide us with more information if this fires again.